mobilecoind-json/src/bin/main.rs

```diff
@@ -478,21 +478,30 @@ fn submit_tx(
 }
 
 /// Checks the status of a transfer given a key image and tombstone block
-#[post("/tx/status-as-sender", format = "json", data = "<receipt>")]
+#[post("/tx/status-as-sender", data = "<receipt>")]
 fn check_transfer_status(
     state: rocket::State<State>,
-    receipt: Json<JsonSendPaymentResponse>,
+    receipt: String,
 ) -> Result<Json<JsonStatusResponse>, String> {
+    // allow receipt to be submitted as either JsonSendPaymentResponse or JsonSenderTxReceipt
```

Comment:
Why is this needed?
Please use `or_else` instead of having two match statements

Reply:
I guess the Sender will always have access to the complete "JsonSendPaymentResponse" so this isn't essential...

It seemed more intuitive to me when scripting `mobilecoind-json` that I would send only the Sender receipt object to this endpoint, since that mirrors what we do on the recipient side.

Reply:
I was also anticipating that we would use something similar to fix [MCC-1894(]https://mobilecoin.atlassian.net/browse/MCC-1894)

Reply:
note that both the gRPC and python library pass only the sender receipt to this function